tfhe/src/core_crypto/algorithms/lwe_programmable_bootstrapping.rs

```diff
@@ -335,6 +480,155 @@ pub fn add_external_product_assign<Scalar, OutputGlweCont, InputGlweCont, GgswCo
     add_external_product_assign_mem_optimized(out, ggsw, glwe, fft, buffers.stack());
 }
 
+#[cfg_attr(__profiling, inline(never))]
+pub(crate) fn collect_next_term<Scalar: UnsignedTorus>(
+    decomposition: &mut TensorSignedDecompositionLendingIter<'_, Scalar>,
+    next_term: &mut [Scalar],
+) -> DecompositionLevel {
+    decomposition.fill_next_term(next_term).unwrap().0
+}
+
+#[cfg_attr(__profiling, inline(never))]
+pub(crate) fn update_with_fmadd(
```

Comment:
I believe the update_with_fmadd could be seen as FFT primitives ? As this allows to make polynomial mulitplications in an efficient way ? what do you think ?

Reply:
makes sense to me. i can put them with the fft